### PR TITLE
SDIT-1826 Add client for NOMIS API

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/prisonperson/PrisonPersonNomisApiService.kt [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+@Service
+class PrisonPersonNomisApiService(@Qualifier("nomisApiWebClient") private val webClient: WebClient) {
+  suspend fun upsertPhysicalAttributes(offenderNo: String, height: Int?, weight: Int?): UpsertPhysicalAttributesResponse =
+    webClient.put()
+      .uri("/prisoners/{offenderNo}/physical-attributes", offenderNo)
+      .bodyValue(UpsertPhysicalAttributesRequest(height, weight))
+      .retrieve()
+      .awaitBody()
+}
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/prisonperson/PrisonPersonNomisApiService.kt:18]: The main change is to implement and test the upsert API.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesReconTest.kt [hunk 2]
```diff
@@ -311,7 +311,7 @@ class ActivitiesReconTest {
       // stub the booking details calls for all will be reported on (that don't match counts)
       bookingCounts
         .filterNot { it.nomisCount == it.dpsCount }
-        .map { PrisonerDetails(it.offenderNo, it.bookingId, it.location) }
+        .map { PrisonerDetails(it.offenderNo, it.bookingId, it.location, true) }
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesReconTest.kt:314]: Unrelated change pulled in from OpenAPI spec update.

**Review by petergphillips — APPROVED** [review]
LGTM
